File: src/android/AndroidLog.java
package org.apache.cordova.log;

import android.util.Log;

import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * This class echoes a string called from JavaScript.
 */
public class Log extends CordovaPlugin {

    @Override
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
        if (action.equals("printLog")) {
            String message = args.getString(0);
            this.printLog(message, callbackContext);
            return true;
        }
        return false;
    }

    private void printLog(String message, CallbackContext callbackContext) {
        if (message != null && message.length() > 0) {
            Log.d("printNative", message);
            callbackContext.success(message);
        } else {
            callbackContext.error("Expected one non-empty string argument.");
        }
    }
}

File: src/android/AndroidLog.kt
package org.apache.cordova.log

import org.apache.cordova.CallbackContext
import org.apache.cordova.CordovaPlugin
import org.json.JSONArray
import org.json.JSONException

/**
 * This class echoes a string called from JavaScript.
 */
class AndroidLog : CordovaPlugin() {
    @Throws(JSONException::class)
    override fun execute(action: String, args: JSONArray, callbackContext: CallbackContext): Boolean {
        val message = args.getString(0)
        if (action == "loadModule") {
            loadModule(message, callbackContext)
            return true
        } else if (action == "pageLoaded") {
            handlePageLoadedMsg(message, callbackContext)
        }
        return false
    }

    private fun loadModule(message: String?, callbackContext: CallbackContext) {
        if (!message.isNullOrEmpty()) {
            val activity = cordova.activity as MultiViewActivity
            activity.loadNewWebView("file:///android_asset/www/test.html")
            callbackContext.success(message)
        } else {
            callbackContext.error("Expected one non-empty string argument.")
        }
    }

    private fun handlePageLoadedMsg(message: String?, callbackContext: CallbackContext) {
        if (!message.isNullOrEmpty()) {
            val activity = cordova.activity as MultiViewActivity
            activity.performViewtransition()
            callbackContext.success(message)
        } else {
            callbackContext.error("Expected one non-empty string argument.")
        }
    }
}